librz/analysis/jmptbl.c

```diff
@@ -134,50 +125,49 @@ RZ_API bool try_walkthrough_casetbl(RzAnalysis *analysis, RzAnalysisFunction *fc
 			}
 		}
 
-		const ut64 jmpptr_idx_off = casetbl_loc + case_idx;
+		const ut64 jmpptr_idx_off = params->casetbl_loc + case_idx;
 		rz_meta_set_data_at(analysis, jmpptr_idx_off, 1);
 		rz_analysis_hint_set_immbase(analysis, jmpptr_idx_off, 10);
 
-		apply_case(analysis, block, ip, sz, jmpptr, case_idx + start_casenum_shift, jmptbl_loc + jmpptr_idx * sz);
-		analyze_new_case(analysis, fcn, block, ip, jmpptr, depth);
+		apply_case(analysis, block, params->ip, params->entry_size, jmpptr, case_idx + params->case_shift, params->jmptbl_loc + jmpptr_idx * params->entry_size);
+		rz_analysis_task_item_new(analysis, params->tasks, fcn, NULL, jmpptr);
 	}
 
 	if (case_idx > 0) {
-		if (default_case == 0) {
-			default_case = UT64_MAX;
+		if (params->default_case == 0) {
+			params->default_case = UT64_MAX;
 		}
-		apply_switch(analysis, ip, jmptbl_loc, case_idx, default_case);
+		apply_switch(analysis, params->ip, params->jmptbl_loc, case_idx, params->default_case);
 	}
 
 	free(jmptbl);
 	free(casetbl);
 	return ret;
 }
 
-RZ_API bool try_walkthrough_jmptbl(RzAnalysis *analysis, RzAnalysisFunction *fcn, RzAnalysisBlock *block, int depth, ut64 ip, st64 start_casenum_shift, ut64 jmptbl_loc, ut64 jmptbl_off, ut64 sz, ut64 jmptbl_size, ut64 default_case, bool ret0) {
-	bool ret = ret0;
-	// jmptbl_size can not always be determined
-	if (jmptbl_size == 0) {
-		jmptbl_size = JMPTBL_MAXSZ;
+RZ_API bool rz_analysis_walkthrough_jmptbl(RzAnalysis *analysis, RzAnalysisFunction *fcn, RzAnalysisBlock *block, RzAnalysisJmpTableParams *params) {
```

Comment:
And here.

Reply:
You also dereference `params` here, please add `RZ_NONNULL` and add a `rz_return_val_if_fail(params)`, same for other parameters, see https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#custom-pointer-modifiers